### PR TITLE
[Refactor] Dependency inject GCS uploader

### PR DESCRIPTION
**Pull request by supasate**
- Create GCSUploader class to be used with photo and video services.
- Dependency inject an uploader into upload function to make it less coupling.
- Add photo test GET and POST (stub GCSUploader for POST).

### CODE REVIEW

**Comment by supasate** [comment]
@parnurzeal Review plz.

File: src/services/photo/index.js [hunk 1]
```diff
@@ -151,6 +53,32 @@ function respondWithPhotoMetadata(photoDocument) {
...
+  if (!uploader.can('upload')) {
+    return Promise.reject(new Error('`upload` function is not available'));
+  }
+
+  return uploader.upload(reqFile)
+  .then((file) => savePhotoMetadata(file))
+  .then((photoDoc) => respondWithPhotoMetadata(photoDoc))
+  .catch((err) => Promise.reject(err));
+}
+
```
Comment (parnurzeal) [inline src/services/photo/index.js:71]: Description. Same as above function.

File: src/services/photo/index.js [hunk 2]
```diff
@@ -151,6 +53,32 @@ function respondWithPhotoMetadata(photoDocument) {
...
+ * @param reqFile a file attached with requested ObjectId
+ * @param uploader a file uploader supporting `upload` function
+ */
+function uploadSaveResponse(reqFile, uploader) {
+  if (!uploader.can('upload')) {
+    return Promise.reject(new Error('`upload` function is not available'));
+  }
+
+  return uploader.upload(reqFile)
+  .then((file) => savePhotoMetadata(file))
```
Comment (parnurzeal) [inline src/services/photo/index.js]: tab

File: src/services/video/index.js [hunk 3]
```diff
@@ -117,8 +70,9 @@ class VideosService {
 
   create(data, params) {
     const gcsConfig = this.app.get('gcs');
+    const gcsUploader = new GCSUploader(gcsConfig);
 
-    return uploadToGCS(params.file, gcsConfig)
+    return gcsUploader.upload(params.file)
     .then((file) => saveVideoMetadata(file))
```
Comment (parnurzeal) [inline src/services/video/index.js]: tab

**Pull request merged**